File: SoccerPlayerApi/SoccerPlayerApi/Dtos/Facts/FactDto.cs
using SoccerPlayerApi.Dtos.Aggregations;
using SoccerPlayerApi.Entities.Structure;

namespace SoccerPlayerApi.Dtos.Facts;

public class FactDto
{
    public int Id { get; set; }

    public int DataTypeId { get; set; }
    public DataType DataType { get; set; }

    public decimal Amount { get; set; }


    public AggregationDto Aggregation1 { get; set; }

    public AggregationDto? Aggregation2 { get; set; }

    public AggregationDto? Aggregation3 { get; set; }

    public AggregationDto? Aggregation4 { get; set; }


    public int TimeAggregationId { get; set; }
    public TimeAggregation TimeAggregation { get; set; }
}

File: SoccerPlayerApi/clientapp/src/app/models/facts/FactDto.ts
import { AggregationDto } from "../aggregations/aggregationDto";
import { DataTypeDto } from "./DataTypeDto";
import { TimeAggregationDto } from "./timeAggregationDto";

export class FactDto {
    id: number;

    dataTypeId: number;
    dataType: DataTypeDto;

    amount: number;

    aggregation1: AggregationDto;

    aggregation2?: AggregationDto;

    aggregation3?: AggregationDto;

    aggregation4?: AggregationDto;

    timeAggregationId: number;
    timeAggregation: TimeAggregationDto;

    constructor(
        id: number,
        dataTypeId: number,
        dataType: DataTypeDto,
        amount: number,
        timeAggregationId: number,
        timeAggregation: TimeAggregationDto,
        aggregation1: AggregationDto,
        aggregation2?: AggregationDto,
        aggregation3?: AggregationDto,
        aggregation4?: AggregationDto
    ) {
        this.id = id;
        this.dataTypeId = dataTypeId;
        this.dataType = dataType;
        this.amount = amount;
        this.aggregation1 = aggregation1;
        this.aggregation2 = aggregation2;
        this.aggregation3 = aggregation3;
        this.aggregation4 = aggregation4;
        this.timeAggregationId = timeAggregationId;
        this.timeAggregation = timeAggregation;
    }
}
